npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -130,12 +130,13 @@ module NpmAndYarn
     YARN_PACKAGE_NOT_FOUND = /MessageError: Couldn't find any versions for "(?<pkg>.*?)" that matches "(?<ver>.*?)"/
 
     YN0001_FILE_NOT_RESOLVED_CODES = T.let({
-      FIND_PACKAGE_LOCATION: /YN0001: UsageError: Couldn't find the (?<pkg>.*) state file/,
-      NO_CANDIDATE_FOUND: /YN0001: Error: (?<pkg>.*): No candidates found/,
-      NO_SUPPORTED_RESOLVER: /YN0001:*.*Error: (?<pkg>.*) isn't supported by any available resolver/,
-      WORKSPACE_NOT_FOUND: /YN0001: Error: (?<pkg>.*): Workspace not found/,
-      ENOENT: /YN0001:*.*Thrown Error: (?<pkg>.*) ENOENT/,
-      MANIFEST_NOT_FOUND: /YN0001: Error: (?<pkg>.*): Manifest not found/
+      FIND_PACKAGE_LOCATION: /YN0001:(.*?)UsageError: Couldn't find the (?<pkg>.*) state file/,
+      NO_CANDIDATE_FOUND: /YN0001:(.*?)Error: (?<pkg>.*): No candidates found/,
+      NO_SUPPORTED_RESOLVER: /YN0001:(.*?)Error: (?<pkg>.*) isn't supported by any available resolver/,
```

Comment:
where are these constants used?

Reply:
@abdulapopoola , they are used in YN0001 class of errors , as YN0001 is not a yarn error but application error, once we capture YN0001 , we evaluate it further to whether raise `DependencyFileNotResolvable/ PrivateSourceAuthenticationFailure`/ parse errors etc. in case we cant find a match , we let it raise `Dependabot::DependabotError`.